crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -1170,7 +1191,11 @@ impl<'src> Parser<'src> {
         // x += *yield from x
         // x += *lambda x: x
         // x += y := 1
-        let value = self.parse_yield_expression_or_else(Parser::parse_star_expression_list);
+        let value = self.parse_expression_list(
+            ExpressionContext::default()
+                .with_yield_expression_allowed()
+                .with_starred_expression_allowed(StarredExpressionPrecedence::BitwiseOr),
+        );
 
```

Comment:
Nit: I would probably have kept `parse_star_expression_list` to avoid the repetition of the context creation (or have a helper that creates the star expression context)

Reply:
Yeah, I'm thinking of going through the `ExpressionContext` creation sites and creating methods for the most common case(s).

Reply:
invocations | count
-- | --
starred, conditional | 5
starred, bitwise | 11
starred, bitwise, yield | 7
others | ...

Based on the above table, I'm adding the following methods:
* `ExpressionContext::starred_conditional`
* `ExpressionContext::starred_bitwise_or`
* `ExpressionContext::yield_or_starred_bitwise_or`